BUILD.tools

```diff
@@ -8,34 +8,25 @@ target(name = 'scala-js-compiler', dependencies=[
 ])
 target(name = 'scala-js-library', dependencies=['contrib/scalajs/3rdparty/jvm/org/scala-js:library'])
 
+
+# TODO: Attempting to use these targets fails because BootstrapJvmTools.get_alternate_target_roots()
+# doesn't descend into dependent target directories and uses the same build_graph it starts with.
+# target(name='scrooge-gen', dependencies=['3rdparty/jvm/com/twitter:scrooge-gen'])
+# target(name='scrooge-linter', dependencies=['3rdparty/jvm/com/twitter:scrooge-linter'])
+
+SCROOGE_REV = '18.12.0'
 # NB: The scrooge tools do not mix their classpaths with the sources they interact with, and
 # therefore they do not need to use the `scala-platform` via scala_jar.
 jar_library(name = 'scrooge-gen',
             jars = [
-              jar(org='com.twitter', name='scrooge-generator_2.11', rev='3.20.0',
-                        excludes=[
-                          # scrooge requires libthrift 0.5.0-1 which is not available on
-                          # the default maven repos. Force scrooge to use thrift-0.6.1, which
-                          # is compatible, instead.
-                          exclude(org = 'org.apache.thrift', name = 'libthrift')
-                        ])
-            ],
-            dependencies = [
-              '3rdparty:thrift-0.6.1',
+              scala_jar(org='com.twitter', name='scrooge-generator', rev=SCROOGE_REV)
```

Comment:
This should not (or at least, "does not") need to use `scala_jar`, because it runs as a tool that generates code, and is never linked into a user's application. This generally won't matter, but with regard to sending the right signal during a pants version upgrade, it's probably better to leave tools that don't need to match the current scala version with their suffix version (`_2.11` in this case) explicitly specified.

Reply:
We use `scala_jar` for `scrooge-core` and `finagle-thrift`, defined in `3rdparty/jvm/com/twitter/BUILD`. Is it safe to use `jar(scrooge-generator_2.11)` here, but use `scala_jar` in that BUILD file?

Reply:
Yes, and necessary. Because those _are_ linked into a user's app.